### PR TITLE
feat(frontend):support batch source node in frontend

### PR DESCRIPTION
**Pull request by ZENOTME**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

summarize this PR :
1. support batch_source node 
2. move SourceInfo from stream_plan.proto to catalog.proto. (Because we need it in the batch plan now
3. when generating the plan stage, try to use  SplitEnumerator to fetch the split info. Use split info to decide the parallelism of the stage containing the batch source. 

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
#6358 #6360

### CODE REVIEW

File: src/frontend/src/optimizer/plan_node/logical_source.rs [hunk 1]
```diff
@@ -102,10 +102,7 @@ impl PredicatePushdown for LogicalSource {
 
 impl ToBatch for LogicalSource {
     fn to_batch(&self) -> Result<PlanRef> {
-        Err(RwError::from(ErrorCode::NotImplemented(
-            "there is no batch source operator".to_string(),
-            None.into(),
-        )))
+        Ok(BatchSource::new(self.clone()).into())
```
Comment (liurenjie1024) [inline src/frontend/src/optimizer/plan_node/logical_source.rs:105]: I would prefer to keep returning here before implementing source executor

File: src/frontend/src/scheduler/plan_fragmenter.rs [hunk 2]
```diff
@@ -189,6 +192,22 @@ impl Query {
     }
 }
 
+#[derive(Clone, Debug)]
+pub struct SourceInfo {
```
Comment (liurenjie1024) [inline src/frontend/src/scheduler/plan_fragmenter.rs:196]:
```suggestion
pub struct SourceScanInfo {
```

File: proto/batch_plan.proto [hunk 3]
```diff
@@ -48,11 +49,14 @@ message ScanRange {
...
-message SourceScanNode {
-  uint32 table_id = 1;
-  // timestamp_ms is used for offset synchronization of high level consumer groups, this field will be deprecated if a more elegant approach is available in the future
-  int64 timestamp_ms = 2;
-  repeated int32 column_ids = 3;
+message SourceNode {
+  uint32 source_id = 1; // use source_id to fetch SourceDesc from local source manager
+  catalog.ColumnIndex row_id_index = 2;
+  repeated plan_common.ColumnCatalog columns = 3;
+  repeated int32 pk_column_ids = 4;
```
[the remark this answers is not in the document]
Comment (ZENOTME) [reply proto/batch_plan.proto:56]: Seems we still need the ColumnIndex. Anyway, it depend on our batch source executor so I think we can remain them now?

[the remark this answers is not in the document]
Comment (liurenjie1024) [reply proto/batch_plan.proto:56]: I don't see why we need `ColumnIndex`, seems it's used for marking column idx for row id?

[the remark this answers is not in the document]
Comment (ZENOTME) [reply proto/batch_plan.proto:56]: Yes, we can remove it now. I find we can implement batch source executor avoiding to use SourceDescBuilder. But we still need source_id to create source_reader and columns to create ConnectSource.

[the remark this answers is not in the document]
Comment (liurenjie1024) [reply proto/batch_plan.proto:56]: OK, let's leave it for now, and remove it later when we confirm that it's unnecessary.

**Review by liurenjie1024 — APPROVED** [review]
Generally LGTM

**Pull request merged**